test/syscalls/linux/kill.cc

```diff
@@ -102,6 +102,62 @@ TEST(KillTest, CanKillAllPIDs) {
   EXPECT_EQ(0, WEXITSTATUS(status));
 }
 
+// reaper process of pid namespace will ignore blockable
+// signal with SIG_DFL action, such as SIGTERM
+TEST(KillTest, CannotTermReaper) {
+  pid_t pid = 1;
+  int status;
+  EXPECT_THAT(kill(pid, SIGTERM), SyscallSucceeds());
+  ASSERT_THAT(RetryEINTR(waitpid)(pid, &status, 0),
+		      SyscallFailsWithErrno(ECHILD));
+}
+
+// nonreaper process of pid namespace can recieve blockable
+// signal with SIG_DFL action, such as SIGTERM
+TEST(KillTest, CanTermNonReaper) {
+  pid_t pid = fork();
+  if (pid == 0) {
+    while (true) {
+      pause();
+    }
+  }
+
+  ASSERT_GT(pid, 0);
+
+  EXPECT_THAT(kill(pid, SIGTERM), SyscallSucceeds());
+
+  int status;
+  ASSERT_THAT(RetryEINTR(waitpid)(pid, &status, 0),
+              SyscallSucceedsWithValue(pid));
+  EXPECT_TRUE(WIFSIGNALED(status));
+  EXPECT_EQ(SIGTERM, WTERMSIG(status));
+}
+
+bool ReaperActed = false;
```

Comment:
Move this inside the test that uses it.

Reply:
var `ReaperActed` and func `ReaperSigHandler` is defined as global var and func, like `SigHandler` defined for case `CanKillAllPIDs`, I didn't think `ReaperSigHandler` cat move into `CanTermReaperAction `, or could you tell me how to do this, I not very familiar with C++, thanks.